pcawg_metadata_parser/fix_metadata_dir/fix_mismatch_metadata_id.py

```diff
@@ -0,0 +1,270 @@
+    return annotations
+
+
+def write_file(flist, fn):
+    with open(fn, 'w') as f:
+        header = True  
+        for r in flist:
+            if header:
+                f.write('\t'.join(r.keys()) + '\n')
+                header = False 
+            # make the list of output from dict
+            line = []
+            for p in r.keys():
+                if isinstance(r.get(p), list):
+                    line.append('|'.join(r.get(p)))
+                elif isinstance(r.get(p), set):
+                    line.append('|'.join(list(r.get(p))))
+                elif r.get(p) is None:
+                    line.append('')
+                else:
+                    line.append(str(r.get(p)))
+            f.write('\t'.join(line) + '\n') 
+
+
+def main(argv=None):
+
+    parser = ArgumentParser(description="Metadata fix",
+             formatter_class=RawDescriptionHelpFormatter)
+    parser.add_argument("-f", "--information for gnos entries", dest="fname",
+             help="Specify file to update the metadata", required=True)
+    parser.add_argument("-o", "--fixed_metadata_dir output folder", dest="fixed_dir",
+             help="Specify output folder for the fixed metadata", required=False)
+
+    args = parser.parse_args()
+    fname = args.fname
+    fixed_dir = args.fixed_dir
+
+
+    if not fixed_dir: fixed_dir = 'fixed_dir'
+    logger.setLevel(logging.INFO)
+    ch.setLevel(logging.WARN)
+
+    log_file = re.sub(r'\.py$', '.log', os.path.basename(__file__))
+
+    fh = logging.FileHandler(log_file)
+    fh.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    fh.setFormatter(formatter)
+    ch.setFormatter(formatter)
+    logger.addHandler(fh)
+    logger.addHandler(ch)
+
+    annotations = {}
+    read_annotations(annotations, 'ESAD-UK', 'ESAD-UK_10broken.txt')
+    read_annotations(annotations, 'mismatch_metadata', 'specimens_with_mismatch_effective_xml_md5sum.txt')
+    # read_annotations(annotations, 'id_mapping', 'ESAD-UK_id_fixes.tsv')
+    # read_annotations(annotations, 'id_mapping', 'PAEN-AU_id_fixes.tsv')
+    read_annotations(annotations, 'id_mapping', 'MELA-AU_PCAWG-DCC_specimen_id_mapping.tsv', 'submitter_specimen_id')
+    read_annotations(annotations, 'id_mapping', 'OV-AU_id_fixes.tsv', 'submitter_specimen_id')
+    read_annotations(annotations, 'id_mapping', 'OV-AU_id_fixes.tsv', 'submitter_sample_id')
+    read_annotations(annotations, 'id_mapping', 'OV-AU_specimen_type_fixes.tsv', 'dcc_specimen_type')
+    read_annotations(annotations, 'id_mapping', 'PAEN-IT_project_code_fixes.tsv', 'dcc_project_code')
+    read_annotations(annotations, 'id_mapping', 'PAEN-AU_project_code_fixes.tsv', 'dcc_project_code')
+    read_annotations(annotations, 'id_mapping', 'ESAD-UK_specimen_type_fixes.tsv', 'dcc_specimen_type')
+    read_annotations(annotations, 'id_mapping', 'LIRI-JP_specimen_id_fixes.tsv', 'submitter_specimen_id')
```

Comment:
are these files checked in git?

I think we should, and also, should put them in a subdirectory.